plugins/nanospell/plugin.js

```diff
@@ -190,7 +190,7 @@
 			// reached the end of block,
 			// so just return what we've walked
 			// of the current word.
-			if (word) return { word: word, range: wordRange };
+			return { word: word, range: wordRange };
```

Comment:
In the course of originally answering your question, I realized that solution as well and added it in.  Between that and changing how the end of the range was set at the end of the `for` loop, it seems to be working quite well!